apps/console/src/features/users/components/user-roles-list.tsx

```diff
@@ -54,17 +62,56 @@ export const UserRolesList: FunctionComponent<UserRoleEditPropsInterface> = (
     const [ activeOption, setActiveOption ] = useState<RolesMemberInterface>(undefined);
     const [ showEmptyRolesListPlaceholder, setShowEmptyRolesListPlaceholder ] = useState<boolean>(false);
 
+    const isGroupAndRoleSeparationEnabled: boolean = useSelector((state: AppState) => 
+        state?.config?.ui?.isGroupAndRoleSeparationEnabled);
+
     /**
      * Set initial selected roles options
      */
     useEffect(() => {
-        if ( user?.roles?.length > 0 ) {
-            setInitialSelectedRolesOptions(user.roles);
+        let userRoles: RolesMemberInterface[];
+        
+        if (isGroupAndRoleSeparationEnabled && user?.roles?.length > 0) {
+            userRoles = user.roles;
+        } else {
+            userRoles = extractUserRolesFromGroups(user?.groups);
+        }
+
+        if (userRoles?.length > 0) {
+            setInitialSelectedRolesOptions(userRoles);
         } else {
             setShowEmptyRolesListPlaceholder(true);
         }
     }, [ user ]);
 
+    /**
+     * When Group and Role Separation is enabled, the groups section of the user will contain both roles and groups.
+     * This method can be used to extract roles from the groups section.
+     * 
+     * @param groups - Groups of the user
+     * @returns Roles of the user
+     */
+    const extractUserRolesFromGroups = (groups: RoleGroupsInterface[]): RolesMemberInterface[] => {
+
+        const userRoles: RolesMemberInterface[] = [];
+        
+        groups?.forEach((group: RoleGroupsInterface) => {
+            const splitDisplayName: string[] = group?.display?.split(DOMAIN_SEPARATOR);
```

Comment:
We can improve readability further as follows.

```suggestion
            const [displayName, userStoreDomain ] = group?.display?.split(DOMAIN_SEPARATOR);
```

just a suggestion :)

Reply:
Resolved with 3741ac80f563720a212d984785f0001795df74d8